File: helloworld.py
"""
需求：在终端输出文本helloworld
流程：
    1.导包  ；
    2.初始化ros2客户端；
    3.创建节点
    4.输出日志
    5.释放资源
"""
#1.导包  
import rclpy

def main():
    #2.初始化ros2客户端；
    rclpy.init()
    #3.创建节点
    node = rclpy.create_node("helloworld_node_py")
    #4.输出日志
    node.get_logger().info("hello world!(python)")
    #5.释放资源
    
    rclpy.shutdown()
    
    
if __name__ == '__main__':
    main()
    

File: helloworld.cpp
/*
需求：在终端输出文本helloworld
流程：
    1.包含头文件；
    2.初始化ros2客户端；
    3.创建节点指针
    4.输出日志
    5.释放资源
    
*/
#include "rclcpp/rclcpp.hpp"
int main(int argc, char **argv){

    //2.初始化ros2客户端；
    rclcpp::init(argc,argv);
    //3.创建节点指针
    auto node = rclcpp::Node::make_shared("helloworld_node_cpp");
    //4.输出日志
    RCLCPP_INFO(node->get_logger(),"hello world");
    //5.释放资源
    rclcpp::shutdown();
return 0;
}
